File: code/boj/1439.js
const filePath =
  process.platform === "linux"
    ? "/dev/stdin"
    : "/Users/jeonsiwon/Desktop/coding-interview/code/boj/input.txt";
const str = require("fs").readFileSync(filePath, "utf-8").toString();

function solution() {
  let zero = 0;
  let one = 0;
  for (let i = 0; i < str.length; i++) {
    if (str[i] != str[i + 1]) {
      if (str[i] === "0") zero += 1;
      else one += 1;
    }
  }
  console.log(Math.min(zero, one));
}
solution();

File: problem/boj/1439.cpp
#include <iostream>
#include <string>
using namespace std;

int zero=0, one=0;
string str;

void solution(){
  cin >> str;
  for(int i=0;i<str.size();i++){ // 범위 
    if(str[i]!=str[i+1]){
      if(str[i]=='0') zero+=1;
      else one+=1;
    }
  }
  cout << min(zero,one);
}

int main(void){
  ios::sync_with_stdio(false);
  cin.tie(0);
  cout.tie(0);
  freopen("input.txt","r",stdin);
  solution(); 
}